src/shared/utils/dataListening.ts

```diff
@@ -0,0 +1,62 @@
+import {runQuery} from 'src/shared/apis/query'
+import {LoadingState} from 'src/homepageExperience/components/ConnectionInformation'
+
+export const TIMEOUT_MILLISECONDS = 60000
+export const TIMER_WAIT = 1000
+
+export const continuouslyCheckForData = (orgID, bucket, updateResponse) => {
+  const startTime = Date.now()
+
+  const intervalId = setInterval(() => {
+    try {
+      const status = checkForData(orgID, bucket)
+      status.then(value => {
+        if (value === LoadingState.Done) {
+          updateResponse(LoadingState.Done)
+          clearInterval(intervalId)
+          return
+        }
+        if (value === LoadingState.Error) {
+          updateResponse(LoadingState.Error)
+          clearInterval(intervalId)
+        }
+        const timePassed = Date.now() - startTime
+
+        if (timePassed > TIMEOUT_MILLISECONDS) {
+          updateResponse(LoadingState.NotFound)
+          clearInterval(intervalId)
+        }
+      })
+    } catch {
+      updateResponse(LoadingState.Error)
+      clearInterval(intervalId)
+    }
```

Comment:
I don't think the `try/catch` will work the way you're expecting here - it requires being in an `async/await` call for it to work like non-asynchronous code. So since this is a more traditional promise using `then`, the `try/catch` should be changed to a `then().catch()` which is identical, functionally. We can also just chain the `then()` and `catch()` together without creating the `status` variable

```suggestion
    checkForData(orgID, bucket).then(value => {
      if (value === LoadingState.Done) {
        updateResponse(LoadingState.Done)
        clearInterval(intervalId)
        return
      }
      if (value === LoadingState.Error) {
        updateResponse(LoadingState.Error)
        clearInterval(intervalId)
      }
      const timePassed = Date.now() - startTime

      if (timePassed > TIMEOUT_MILLISECONDS) {
        updateResponse(LoadingState.NotFound)
        clearInterval(intervalId)
      }
    }).catch(() => {
      updateResponse(LoadingState.Error)
      clearInterval(intervalId)
    })
```

Reply:
This makes sense, I added the change and it works !